From 615b2bb4de0bfad294260eea8bdc8658315d92b5 Mon Sep 17 00:00:00 2001
From: Xiaolei Wang <xiaolei.wang@windriver.com>
Date: Wed, 3 Jun 2020 13:32:17 +0800
Subject: [PATCH] Bluetooth: hci_ll: add a small delay for wl12xx enable bt_en

For am335xevmsk, when the wl12xx bt_en pin is enabled, wl12xx cannot
respond in time, resulting in the inability to communicate with the
Bluetooth device. A small delay is required to use Bluetooth normally

Signed-off-by: Xiaolei Wang <xiaolei.wang@windriver.com>
---
 drivers/bluetooth/hci_ll.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/bluetooth/hci_ll.c b/drivers/bluetooth/hci_ll.c
index c04f5f9e1ed0..a16854d7c5c8 100644
--- a/drivers/bluetooth/hci_ll.c
+++ b/drivers/bluetooth/hci_ll.c
@@ -606,6 +606,7 @@ static int ll_setup(struct hci_uart *hu)
 		gpiod_set_value_cansleep(lldev->enable_gpio, 0);
 		msleep(5);
 		gpiod_set_value_cansleep(lldev->enable_gpio, 1);
+		mdelay(100);
 		err = serdev_device_wait_for_cts(serdev, true, 200);
 		if (err) {
 			bt_dev_err(hu->hdev, "Failed to get CTS");
-- 
2.17.1

